firebase-common/src/main/java/com/google/firebase/Timestamp.kt

```diff
@@ -74,19 +72,19 @@ class Timestamp : Comparable<Timestamp>, Parcelable {
     compareValuesBy(this, other, Timestamp::seconds, Timestamp::nanoseconds)
 
   override fun equals(other: Any?): Boolean =
-    when {
-      (other === this) -> true
-      (other !is Timestamp) -> false
-      else -> compareTo(other) == 0
-    }
+    other == this || other is Timestamp && compareTo(other) == 0
 
-  override fun hashCode(): Int = 31 * seconds.hashCode() + nanoseconds
+  override fun hashCode(): Int {
```

Comment:
How about just using `Objects.hash()` here?

https://developer.android.com/reference/java/util/Objects#hash(java.lang.Object[])

Reply:
We'd rather not introduce behavioural changes unless strongly needed, as bugs introduced by these changes are extremely hard to find.